### PR TITLE
Simulator inital draft

### PR DESCRIPTION
**Pull request by 0x4d**
## Changes
- Added event handler listening for registered simulators
- Store available simulators
- Added menu entries for simulators

## TODO
- Activate simulator on click (and add visual indication)

### CODE REVIEW

File: src/components/header/menu/dropdowns/MenuDropdownRun.vue [hunk 1]
```diff
@@ -20,6 +20,14 @@
       :shortkey="['ctrl', 'alt', 't']"
       @action="stopSimulation"
     />
+    <MenuDropdownSeparator
+      v-if="simulators && simulators.length"
+    />
+    <MenuDropdownItem
```
Comment (tim-kilian) [inline src/components/header/menu/dropdowns/MenuDropdownRun.vue:30]: Are each simulators separate menuItems?

Comment (0x4d) [reply src/components/header/menu/dropdowns/MenuDropdownRun.vue:30]: Yes. Maybe this should rather be called `formalisms` as a simulator can provide different formalisms?

Comment (tim-kilian) [reply src/components/header/menu/dropdowns/MenuDropdownRun.vue:30]: Yes, that sounds better, renew is also letting you choose your formalism that you want to simulate.

Comment (0x4d) [reply src/components/header/menu/dropdowns/MenuDropdownRun.vue:30]: It's just semantics in the end. The way it is now, there is one simulator for one formalism and a gateway plugin can provide multiple simulators (that may or may not use the same backend).

**Review by tim-kilian — APPROVED** [review]
Looks solid for me. Can this be merged now? I would like to remove the WIP.

**Comment by 0x4d** [comment]
Yep yep yep!

**Pull request merged**